dlux_global_planner/CMakeLists.txt

```diff
@@ -35,14 +35,15 @@ add_library(dgp src/dlux_global_planner.cpp src/cost_interpreter.cpp)
 add_dependencies(dgp ${catkin_EXPORTED_TARGETS})
 target_link_libraries(dgp ${catkin_LIBRARIES})
 
-add_executable(planner_node src/planner_node.cpp)
-add_dependencies(planner_node ${catkin_EXPORTED_TARGETS})
-target_link_libraries(planner_node dgp ${catkin_LIBRARIES})
+add_executable(${PROJECT_NAME}_planner_node src/planner_node.cpp)
```

Comment:
I've renamed the CMake target to `${PROJECT_NAME}_planner_node`, but then the `set_target_properties` line a bit further down renames the output executable back to `planner_node`. The resulting node name doesn't change, only the CMake internals. I've stolen that trick from the default CMakeLists.txt template that gets generated when you run `catkin_create_pkg` (which is my go-to reference for how a Catkin CMakeLists should look like, it even has comments and stuff that explain this trick).

I didn't do the same for the `utest` executables, because they are only used during testing.

Reply:
:+1: 

Apologies for not seeing that the first time. Man I dislike CMake. But good tip on the autogenerated one.